src/app/routes/registration/enrolment-form/enrolment-form.component.ts

```diff
@@ -121,18 +134,31 @@ export class EnrolmentFormComponent implements EnrolmentForm {
 
   private buildEnrolmentFormFields() {
     const values = this.enrolmentForm.value.fields;
-    return this.fieldList.map((field, index) => ({
-      key: field.label,
-      value: field.schema ? JSON.stringify(values[index]) : values[index],
-    }));
+    return this.fieldList.map((field: IFieldDefinition, index) => {
+      if (field.schema) {
+        this.checkSchema(field.schema, {}, field.label);
```

Comment:
Why the `checkSchema` here? I suppose it doesn't hurt but I'm not sure that `this.isValidSchema` is checked after this? Is rationale that an exception will be thrown if the check fails?

Reply:
I'm checking schema at start, because:
1. I saw that at start approve button is enabled even if the field requires array but gets object. When i change that to array and again to object, approve button is disabled. So we need to check validation while setting up.
2. without this `isValidSchema` object won't work properly as we are creating this object while creating fields.